src/apps/relay/dbdrivers/dbd_sqlite.c

```diff
@@ -212,752 +206,799 @@ static sqlite3 *get_sqlite_connection(void) {
+    TURN_LOG_FUNC(
+        TURN_LOG_LEVEL_ERROR,
           "Cannot open SQLite DB connection: <%s>, runtime error:\n  %s\n  (If your intention is to use an SQLite "
           "database for the TURN server, then\n  check and fix, if necessary, the effective permissions of the TURN "
           "server\n  process and of the DB directory and then re-start the TURN server)\n",
           pud->userdb_sanitized, errmsg);
-      if (sqliteconnection) {
-        sqlite3_close(sqliteconnection);
-        sqliteconnection = NULL;
-      }
-      turn_params.default_users_db.userdb_type = TURN_USERDB_TYPE_UNKNOWN;
-    } else {
-      init_sqlite_database(sqliteconnection);
-      if (!donot_print_connection_success) {
-        TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "SQLite DB connection success: %s\n", pud->userdb_sanitized);
-        donot_print_connection_success = 1;
-      }
+    if (sqliteconnection != NULL) {
+      sqlite3_close(sqliteconnection);
+      sqliteconnection = NULL;
     }
-    if (sqliteconnection) {
-      (void)pthread_setspecific(connection_key, sqliteconnection);
+    turn_params.default_users_db.userdb_type = TURN_USERDB_TYPE_UNKNOWN;
+  } else {
+    init_sqlite_database(sqliteconnection);
+    if (!donot_print_connection_success) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "SQLite DB connection success: %s\n", pud->userdb_sanitized);
+      donot_print_connection_success = 1;
     }
   }
+  if (sqliteconnection) {
+    (void)pthread_setspecific(connection_key, sqliteconnection);
+  }
+
   return sqliteconnection;
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 static int sqlite_get_auth_secrets(secrets_list_t *sl, uint8_t *realm) {
   int ret = -1;
-  sqlite3 *sqliteconnection = get_sqlite_connection();
-  if (sqliteconnection) {
-    char statement[TURN_LONG_STRING_SIZE];
-    sqlite3_stmt *st = NULL;
-    int rc = 0;
-    snprintf(statement, sizeof(statement) - 1, "select value from turn_secret where realm='%s'", realm);
 
-    sqlite_lock(0);
+  sqlite3 * const sqliteconnection = get_sqlite_connection();
+  if (sqliteconnection == NULL) {
+    return ret;
+  }
 
-    if ((rc = sqlite3_prepare(sqliteconnection, statement, -1, &st, 0)) == SQLITE_OK) {
+  char statement[TURN_LONG_STRING_SIZE];
+  snprintf(statement, sizeof(statement) - 1, "select value from turn_secret where realm='%s'", realm);
 
-      int ctotal = sqlite3_column_count(st);
-      ret = 0;
+  sqlite_lock(0);
 
-      while (ctotal > 0) {
+  sqlite3_stmt *st = NULL;
+  if (sqlite3_prepare(sqliteconnection, statement, -1, &st, 0) == SQLITE_OK) {
 
-        int res = sqlite3_step(st);
-        if (res == SQLITE_ROW) {
+	ret = 0;
+    if (sqlite3_column_count(st) > 0) {
 
-          int type = sqlite3_column_type(st, 0);
-          if (type != SQLITE_NULL)
-            add_to_secrets_list(sl, (const char *)sqlite3_column_text(st, 0));
+      for(int stepResult = sqlite3_step(st);
```

Comment:
Use a for loop over the result of sqlite3_step() instead of a while-loop over the non-changing `ctotal` (the number of columns) variable.

While the control flow wasn't necessarily... "wrong" before, this is more idomatic.